frappe/website/doctype/web_form/web_form.py

```diff
@@ -425,39 +430,51 @@ def accept(web_form, data, docname=None, for_payment=False):
 		if web_form.login_required and frappe.session.user=="Guest":
 			frappe.throw(_("You must login to submit this form"))
 
-		ignore_mandatory = True if files else False
+		ignore_mandatory = True if base64_files else False
 
 		doc.insert(ignore_permissions = True, ignore_mandatory = ignore_mandatory)
 
-	# add files
-	if files:
-		for f in files:
-			fieldname, filedata = f
-
-			# remove earlier attached file (if exists)
-			if doc.get(fieldname):
-				remove_file_by_url(doc.get(fieldname), doctype=doc.doctype, name=doc.name)
-
-			# save new file
-			filename, dataurl = filedata.split(',', 1)
-			_file = frappe.get_doc({
-				"doctype": "File",
-				"file_name": filename,
-				"attached_to_doctype": doc.doctype,
-				"attached_to_name": doc.name,
-				"content": dataurl,
-				"decode": True})
-			_file.save()
-
-			# update values
-			doc.set(fieldname, _file.file_url)
-
+	# add base64 files
+	for f in base64_files:
+		fieldname, filedata = f
+
+		# remove earlier attached file (if exists)
+		if doc.get(fieldname):
+			remove_file_by_url(doc.get(fieldname), doctype=doc.doctype, name=doc.name)
+
+		# save new file
+		filename, dataurl = filedata.split(',', 1)
+		_file = frappe.get_doc({
+			"doctype": "File",
+			"file_name": filename,
+			"attached_to_doctype": doc.doctype,
+			"attached_to_name": doc.name,
+			"content": dataurl,
+			"decode": True})
+		_file.save()
+
+		# update values
+		doc.set(fieldname, _file.file_url)
 		doc.save(ignore_permissions = True)
 
-	if files_to_delete:
-		for f in files_to_delete:
-			if f:
-				remove_file_by_url(f, doctype=doc.doctype, name=doc.name)
+	for f in files:
+		f = frappe._dict(f)
+		if f.file_url not in files_to_keep:
```

Comment:
Why accept `files` from the client and loop over it when we are just saving the files in the `files_to_keep` array (which is built from the `Attach` and `Attach Image` field values).